bin/deploy-vhost

```diff
@@ -133,140 +133,56 @@ sub get_pgpw_string {
-            $db_ro_host_line = "$params->{prefix}_DB_RO_HOST: '$params->{replica}.ukcod.org.uk'";
-            # Hot standby usually on 5433, unless it's been failed over at some point.
-            if ($params->{port} eq '5432') {
-                $port_ro_line = "$params->{prefix}_DB_RO_PORT: 5433";
-            }
-            else {
-                $port_ro_line = "$params->{prefix}_DB_RO_PORT: 5432";
-            }
-        }
-    }
-    my $db_host = get_db_host($params, $internal);
-    my $pgpw_string = get_pgpw_string($params, $database);
-
-    my $conf = <<END;
-$params->{prefix}_DB_HOST: '$db_host'
-$port_line
-$db_ro_host_line
-$port_ro_line
-$params->{prefix}_DB_NAME: '$database'
-$params->{prefix}_DB_USER: '$database'
-$params->{prefix}_DB_PASS: '\${\\(pgpw('$pgpw_string'))}'
-END
-    return $conf;
-
-}
 
+# For each database, pull in information from the JSON and derive some
+# further data as required. Store this in $conf->{database_configs}
 sub db_conf {
     my ($params, $database) = @_;
     if ($params->{type} ne 'mysql' && $params->{type} ne 'psql' && $params->{type} ne 'mongo'){
         die "unknown database type '$params->{type}'";
     }
 
-    # Strings for PHP configuration
-    $conf->{database_configs}{default} .= one_db_conf_php($params, $database, 1);
-    $conf->{database_configs}{external} .= one_db_conf_php($params, $database, 0);
-
-    # Strings for YAML configuration
-    $conf->{database_configs}{yaml} .= one_db_conf_yml($params, $database, 1);
-    $conf->{database_configs}{external_yaml} .= one_db_conf_yml($params, $database, 0);
-
     my $pgpw_string = get_pgpw_string($params, $database);
 
-    # Strings for Rails configuration
-    if ($params->{type} eq 'psql') {
-            # XXX doesn't cope with multiple Rails database configs (as you
-            # have to list them in the yml file in right place, so one text
-            # option can't), oh well
-            if (scalar(@{$conf->{'databases'}}) == 1) {
-                my $db_host = get_db_host($params, $internal_network);
-                $conf->{database_configs}{rails} = <<END;
-    adapter: postgresql
-    database: $database
-    username: $database
-    password: \${\\(pgpw('$pgpw_string'))}
-    host: $db_host
-    port: $params->{port}
-END
+    my $db_config = {
+        prefix   => $params->{prefix},
+        host     => get_db_host($params, 0),
+        name     => $database,
+        username => $database,
+        password => "\${\\(pgpw('$pgpw_string'))}",
+    };
+
+    if ($params->{port}) {
+        $db_config->{port} = $params->{port};
+        if ($params->{replica}) {
+            $db_config->{replica_host} = "$params->{replica}.ukcod.org.uk";
+            if ($params->{port} eq '5432') {
+                $db_config->{replica_port} = 5433;
+            }
+            elsif ($params->{port} eq '5433') {
+                $db_config->{replica_port} = 5432;
+            }
+            elsif ($params->{port} eq '3306') {
```

Comment:
If you specify replica for a mysql, but don't specify the default port (as you haven't had to do that in the past for mysql), this won't work; should the mysql `elsif` a bit down do a check for replica, just to cover all the bases? Or just make port mandatory then don't need that elsif...